homeassistant/components/cover/garadget.py

```diff
@@ -0,0 +1,271 @@
+    CONF_ACCESS_TOKEN, CONF_NAME, STATE_UNKNOWN, STATE_CLOSED, STATE_OPEN,\
+    CONF_COVERS, ATTR_HIDDEN
+import homeassistant.helpers.config_validation as cv
+
+DEFAULT_NAME = 'Garadget'
+
+ATTR_SIGNAL_STRENGTH = "wifi signal strength (dB)"
+ATTR_TIME_IN_STATE = "time in state"
+ATTR_SENSOR_STRENGTH = "sensor reflection rate"
+
+STATE_OPENING = "opening"
+STATE_CLOSING = "closing"
+STATE_STOPPED = "stopped"
+STATE_OFFLINE = "offline"
+
+STATES_MAP = {
+    "open": STATE_OPEN,
+    "opening": STATE_OPENING,
+    "closed": STATE_CLOSED,
+    "closing": STATE_CLOSING,
+    "stopped": STATE_STOPPED
+}
+
+
+# Validation of the user's configuration
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_DEVICE): cv.string,
+    vol.Optional(CONF_USERNAME): cv.string,
+    vol.Optional(CONF_PASSWORD): cv.string,
+    vol.Optional(CONF_ACCESS_TOKEN): cv.string,
+    vol.Optional(CONF_NAME): cv.string
+})
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the Demo covers."""
+    covers = []
+    covers_conf = config.get(CONF_COVERS)
+
+    for cover in covers_conf:
+        args = {
+            "name": cover.get(CONF_NAME),
+            "device_id": cover.get(CONF_DEVICE),
+            "username": cover.get(CONF_USERNAME),
+            "password": cover.get(CONF_PASSWORD),
+            "access_token": cover.get(CONF_ACCESS_TOKEN)
+        }
+
+        covers.append(GaradgetCover(hass, args))
+
+    add_devices(covers)
+
+
+class GaradgetCover(CoverDevice):
+    """Representation of a demo cover."""
+
+    # pylint: disable=no-self-use, too-many-instance-attributes
+    def __init__(self, hass, args):
+        """Initialize the cover."""
+        self.particle_url = 'https://api.particle.io'
+        self.hass = hass
+        self._name = args['name']
+        self.device_id = args['device_id']
+        self.access_token = args['access_token']
+        self.obtained_token = False
+        self._username = args['username']
+        self._password = args['password']
+        self._state = STATE_UNKNOWN
+        self.time_in_state = None
+        self.signal = None
+        self.sensor = None
+        self._unsub_listener_cover = None
+        self._hidden = False
+
+        if self.access_token is None:
+            self.access_token = self.get_token()
+            self._obtained_token = True
+
+        # Lets try to get the configured name if not provided.
+        try:
+            if self._name is None:
+                doorconfig = self._get_variable("doorConfig")
+                if doorconfig["nme"] is not None:
+                    self._name = doorconfig["nme"]
+            self.update()
+        except requests.exceptions.ConnectionError as ex:
+            _LOGGER.error('Unable to connect to server: %(reason)s',
+                          dict(reason=ex))
+            self._state = STATE_OFFLINE
+            self._hidden = True
```

Comment:
Setting hidden to true is wrong here. The device is just offline - it will be confusing for users if it is hidden all of a sudden. Instead, use [the `available` property](https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/helpers/entity.py#L136-L139). In that case HA will take care of showing that the device is offline.